lib/trino-filesystem/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java

```diff
@@ -53,7 +54,9 @@ public TrinoInput newInput()
             throws IOException
     {
         FileSystem fileSystem = environment.getFileSystem(context, file);
+        long fileOpenStart = System.nanoTime();
         FSDataInputStream input = environment.doAs(context.getIdentity(), () -> fileSystem.open(file));
+        fileOpenTime = System.nanoTime() - fileOpenStart;
```

Comment:
Also I am looking into using `Ticker/Stopwatch` to measure the elapsed time here. Potentially `HdfsFileSystem` can contain a `Ticker` object.

Reply:
^Not necessary.